File: Java/01/tasks-manager-application/src/main/java/org/tasksmanager/DoneTasksEventHandler.java
package org.tasksmanager;

import java.util.logging.Level;
import java.util.logging.Logger;
import javax.inject.Named;
import org.commons.Services.TaskDao;
import org.commons.Services.TasksService;
import org.commons.Tasks.TaskCompleted;
import org.cqrs.Bus;
import org.cqrs.MessageHandler;
import org.cqrs.Repository;
import org.tasksmanager.Repositories.DoneTaskDao;

@Named("doneTasksEventHandler")
public class DoneTasksEventHandler implements MessageHandler {

    private static final Logger logger = Logger.getLogger(DoneTasksEventHandler.class.getSimpleName());
    private final TasksService _tasksService;
    private final Repository<DoneTaskDao> _repository;
    private Bus _bus;

    @Override
    public void Register(Bus bus) {
        _bus = bus;
        _bus.RegisterHandler(m -> Handle((TaskCompleted) m), TaskCompleted.class);
    }

    public DoneTasksEventHandler(TasksService tasksService, Repository<DoneTaskDao> doneTasksRepository) {
        _repository = doneTasksRepository;
        _tasksService = tasksService;
    }

    public void Handle(TaskCompleted message) {
        logger.log(Level.INFO, "TaskCompleted");
        TaskDao taskDao = _tasksService.GetById(message.getId());
        DoneTaskDao doneTask = new DoneTaskDao();
        doneTask.setId(taskDao.getId());
        doneTask.setPriority(taskDao.getPriority());
        doneTask.setCreationDate(taskDao.getCreationDate());
        doneTask.setTitle(taskDao.getTitle());
        doneTask.setCompletionDate(taskDao.getCompletionDate());
        _repository.Save(doneTask);
    }
}

File: Net/02/TasksManager.Application/DoneTasksEventHandler.cs
using Commons.Services;
using Commons.Tasks;
using Cqrs;
using log4net;
using System;
using TasksManager.Repositories;

namespace TasksManager
{
    public class DoneTasksEventHandler : IMessageHandler
    {
        private static ILog _logger = LogManager.GetLogger(typeof(DoneTasksEventHandler));
        private ITasksService _tasksService;
        private IRepository<DoneTaskDao, Guid> _repository;
        private IBus _bus;

        public void Register(IBus bus)
        {
            _bus = bus;
            _bus.RegisterHandler(m => Handle((TaskCompleted)m), typeof(TaskCompleted));
        }

        public DoneTasksEventHandler(ITasksService tasksService, IRepository<DoneTaskDao, Guid> repository)
        {
            _repository = repository;
            _tasksService = tasksService;
        }

        public void Handle(TaskCompleted message)
        {
            _logger.Info("TaskCompleted");
            var taskDao = _tasksService.GetById(message.Id);
            var doneTask = new DoneTaskDao
            {
                Id = taskDao.Id,
                Priority = taskDao.Priority,
                CreationDate = taskDao.CreationDate,
                Title = taskDao.Title,
                CompletionDate = message.CompletionDate
            };
            _repository.Save(doneTask);
        }
    }
}
